CLI login error / panic: runtime error: invalid memory address or nil pointer dereference

### Summary

Getting the title error while trying to login using command:
_**ttn-lw-cli login --api-key $TTS_DEFAULT_CLI_API_KEY**_

This looks like new bug in release 3.34.1, because **I get no error while using 3.33.0 version**.

I use TTS as docker contenerized and locally deployed LoRaWAN network. 

I follow every step of tutorial: https://www.thethingsindustries.com/docs/enterprise/docker/running-the-stack/

To additional automization I create CLI API key using: 
_**docker compose run --rm stack is-db create-user-api-key**_ , 
then I use that key to login, create default application, application API key etc.


### Steps to Reproduce

1. docker compose pull
2. docker compose run --rm stack is-db migrate
3. docker compose run --rm stack is-db create-admin-user \
  --id admin \
  --email your@email.com
4. docker compose run --rm stack is-db create-oauth-client \
  --id cli \
  --name "Command Line Interface" \
  --owner admin \
  --no-secret \
  --redirect-uri "local-callback" \
  --redirect-uri "code"
5. docker compose run --rm stack is-db create-oauth-client \
  --id ${ID} \
  --name "${NAME}" \
  --owner admin \
  --secret "${CLIENT_SECRET}" \
  --redirect-uri "${REDIRECT_URI}" \
  --redirect-uri "${REDIRECT_PATH}" \
  --logout-redirect-uri "${LOGOUT_REDIRECT_URI}" \
  --logout-redirect-uri "${LOGOUT_REDIRECT_PATH}"
6. docker compose run --rm stack is-db create-user-api-key \
    --user-id admin
    --name "cli default API key"
7. docker exec stack ttn-lw-cli login \
    --api-key <cli_default_API_key> \
    --insecure



### Current Result

Login fail, further actions with API KEY cannot be done, cause of the lack of the authentication

### Expected Result

Login should succed, then command should be corectly authenticted:
_INFO  (4.8) Creating default API key for application 'default-app'...
INFO    Telemetry is enabled. Check the documentation for more information on what is collected and how to disable it {"documentation_url": "https://www.thethingsindustries.com/docs/reference/telemetry/cli"}
WARN    Using insecure connection to OAuth server
WARN    Using insecure connection to API
INFO    API key ID: <censored>
INFO    API key value: <censored>_

### Relevant Logs

```shell
WARN    Using insecure connection to OAuth server
WARN    Using insecure connection to API
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x2478750]

goroutine 80 [running]:
go.thethings.network/lorawan-stack/v3/pkg/telemetry/exporter/cli.(*cliTask).Run(0xc000f9cbd0, {0x3255988?, 0xc000f85080?})
        /home/runner/work/lorawan-stack/lorawan-stack/pkg/telemetry/exporter/cli/cli.go:171 +0xf0
go.thethings.network/lorawan-stack/v3/cmd/ttn-lw-cli/commands.init.preRun.func239.2({0x3255988, 0xc000f85080})
        /home/runner/work/lorawan-stack/lorawan-stack/cmd/ttn-lw-cli/commands/root.go:184 +0xa2
created by go.thethings.network/lorawan-stack/v3/cmd/ttn-lw-cli/commands.init.preRun.func239 in goroutine 1
        /home/runner/work/lorawan-stack/lorawan-stack/cmd/ttn-lw-cli/commands/root.go:180 +0x559
```

### URL

_No response_

### Deployment

The Things Stack Open Source (self-hosted)

### The Things Stack Version

3.34.1

### Client Name and Version

```shell
$ ttn-lw-cli version
The Things Network Command-line Interface: ttn-lw-cli
Version:             3.34.1
Build date:          2025-04-28T08:37:25Z
Git commit:          e22b9512d
Go version:          go1.23.3
OS/Arch:             linux/amd64
```

### Other Information

_No response_

### Proposed Fix

_No response_

### Contributing

- [ ] I can help by doing more research.
- [ ] I can help by implementing a fix after the proposal above is approved.
- [ ] I can help by testing the fix before it's released.

### Validation

- [ ] The fix is tested in a staging environment.
- [ ] The fix is documented in [The Things Stack Documentation](https://www.thethingsindustries.com/docs/)

### Code of Conduct

- [x] I agree to follow TTN's [Community Code of Conduct](https://github.com/TheThingsNetwork/Manifesto/blob/master/CodeOfConduct.md).

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -27,6 +27,7 @@ For details about compatibility between different releases, see the **Commitment
 ### Fixed
 
 - Fixed searching users in User management in the Console.
+- CLI no longer causes a panic when running from a docker container in a Linux machine.
 
 ### Security
 

--- a/Dockerfile
+++ b/Dockerfile
@@ -14,7 +14,10 @@ RUN rm -rf /srv/ttn-lorawan/lorawan-webhook-templates/.git
 
 FROM alpine:3.21
 
-RUN addgroup -g 886 thethings && adduser -u 886 -S -G thethings thethings
+RUN addgroup -g 886 thethings && \
+    adduser -u 886 -S -G thethings thethings && \
+    mkdir -p /home/thethings && \
+    chown thethings:thethings /home/thethings
 
 RUN apk --update --no-cache add ca-certificates curl
 
@@ -40,5 +43,8 @@ VOLUME ["/srv/ttn-lorawan/public/blob"]
 ENV TTN_LW_HEALTHCHECK_URL=http://localhost:1885/healthz
 
 HEALTHCHECK --interval=1m --timeout=5s CMD curl -f $TTN_LW_HEALTHCHECK_URL || exit 1
+
+RUN mkdir -p /home/thethings/.cache && chown -R thethings:thethings /home/thethings/.cache
+ENV XDG_CACHE_HOME=/home/thethings/.cache
 
 USER thethings:thethings

--- a/pkg/telemetry/exporter/cli/cli.go
+++ b/pkg/telemetry/exporter/cli/cli.go
@@ -79,50 +79,50 @@ func cliStatePath() (string, error) {
 }
 
 // getCLIState returns the telemetry state, if it doesn't exist it creates a file with default values for the state.
-func getCLIState() (*cliStateTelemetry, bool, error) {
+func getCLIState() (*cliStateTelemetry, error) {
 	folderPath, err := ttnPath()
 	if err != nil {
-		return nil, false, err
+		return nil, err
 	}
 	if _, err := os.Stat(folderPath); err != nil {
 		if !os.IsNotExist(err) {
-			return nil, false, err
+			return nil, err
 		}
 		// Creates the folder since it does not exist.
 		if err := os.Mkdir(folderPath, 0o755); err != nil {
-			return nil, false, err
+			return nil, err
 		}
 	}
 
 	statePath, err := cliStatePath()
 	if err != nil {
-		return nil, false, err
+		return nil, err
 	}
 
 	cliState := &cliStateTelemetry{}
 
 	if _, err := os.Stat(statePath); err != nil {
 		if !os.IsNotExist(err) {
-			return nil, false, err
+			return nil, err
 		}
 
 		cliState.UID = uuid.NewString()
 		cliState.LastSent = time.Now()
-		return cliState, true, nil
+		return cliState, nil
 	}
 
 	b, err := os.ReadFile(statePath)
 	if err != nil {
-		return nil, false, err
+		return nil, err
 	}
 	if err := yaml.Unmarshal(b, cliState); err != nil {
-		return nil, false, err
+		return nil, err
 	}
-	return cliState, false, nil
+	return cliState, nil
 }
 
-func shouldSendTelemetry(t time.Time) bool {
-	return time.Now().After(t.Add(defaultTimeout))
+func shouldSendTelemetry(st *cliStateTelemetry) bool {
+	return st != nil && time.Now().After(st.LastSent.Add(defaultTimeout))
 }
 
 type cliTask struct {
@@ -163,18 +163,23 @@ func NewCLITelemetry(opts ...Option) Task {
 // Run a small task that send telemetry data once a day.
 func (ct *cliTask) Run(ctx context.Context) {
 	logger := log.FromContext(ctx)
-	state, send, err := getCLIState()
+	state, err := getCLIState()
 	if err != nil {
 		logger.WithError(err).Debug("Failed to retrieve telemetry state")
+		return // Skip the telemetry procedure if an error is found creating or fetching the previous state.
 	}
 
-	if send || shouldSendTelemetry(state.LastSent) {
+	if shouldSendTelemetry(state) {
 		state.LastSent = time.Now()
-		if statePath, err := cliStatePath(); err != nil {
+
+		statePath, err := cliStatePath()
+		if err != nil {
 			logger.WithError(err).Debug("Failed to retrieve telemetry state file path")
-		} else {
-			// If no error fetching the state file path, update the last sent timestamp.
-			state.Write(statePath) //nolint:errcheck
+			return
+		}
+		if err := state.Write(statePath); err != nil {
+			logger.WithError(err).Debug("Failed to write new telemetry state")
+			return
 		}
 
 		b, err := json.Marshal(&models.TelemetryMessage{
@@ -186,6 +191,7 @@ func (ct *cliTask) Run(ctx context.Context) {
 			logger.WithError(err).Debug("Failed to marshal telemetry information")
 			return
 		}
+
 		resp, err := http.DefaultClient.Post(ct.target, "application/json", bytes.NewReader(b))
 		if err != nil {
 			logger.WithError(err).Debug("Failed to send telemetry information")
@@ -193,6 +199,7 @@ func (ct *cliTask) Run(ctx context.Context) {
 		}
 		defer resp.Body.Close()
 		io.Copy(io.Discard, resp.Body) // nolint:errcheck
+
 		logger.Info("Sent telemetry information")
 	}
 }

